src/WebJobs.Script/Workers/Http/HttpWorkerFunctionProvider.cs

```diff
@@ -0,0 +1,184 @@
+        private readonly ILogger _logger;
+        private readonly Dictionary<string, List<string>> _errors = new();
+        private static readonly ImmutableArray<string> AllHttpMethods = ["get", "post", "put", "delete", "patch", "head", "options"];
+
+        public HttpWorkerFunctionProvider(IOptions<HttpWorkerOptions> httpWorkerOptions, IOptionsMonitor<LanguageWorkerOptions> languageWorkerOptions, IHostFunctionMetadataProvider hostFunctionMetadataProvider, IEnvironment environment, ILogger<HttpWorkerFunctionProvider> logger)
+        {
+            _httpWorkerOptions = httpWorkerOptions?.Value ?? throw new ArgumentNullException(nameof(httpWorkerOptions));
+            _hostFunctionMetadataProvider = hostFunctionMetadataProvider ?? throw new ArgumentNullException(nameof(hostFunctionMetadataProvider));
+            _languageWorkerOptions = languageWorkerOptions ?? throw new ArgumentNullException(nameof(languageWorkerOptions));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public ImmutableDictionary<string, ImmutableArray<string>> FunctionErrors =>
+           _errors.ToImmutableDictionary(k => k.Key, v => v.Value.ToImmutableArray());
+
+        public async Task<ImmutableArray<FunctionMetadata>> GetFunctionMetadataAsync()
+        {
+            if (!string.Equals(_httpWorkerOptions.WorkerRuntime, ScriptConstants.CustomHandlerWorkerRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                return [];
+            }
+
+            var routes = _httpWorkerOptions.HttpRoutes;
+            if (routes is null || !routes.Any())
+            {
+                return [];
+            }
+
+            var hostFunctionMetadata = await _hostFunctionMetadataProvider.GetFunctionMetadataAsync(_languageWorkerOptions.CurrentValue.WorkerConfigs, forceRefresh: false);
+
+            // We already know custom handler http routes are configured, if function.json files are also present we cannot proceed.
+            if (hostFunctionMetadata.Any())
+            {
+                throw new InvalidOperationException(
+                    "Detected both function.json files and custom handler HTTP route configuration definition in host.json" +
+                    "Only one configuration source is supported. Remove either the function.json files or the HTTP routes entries in host.json.");
+            }
+
+            return CreateFunctionsFromRoutes(routes);
+        }
+
+        private ImmutableArray<FunctionMetadata> CreateFunctionsFromRoutes(IEnumerable<HttpWorkerRoute> routes)
+        {
+            var functions = new Collection<FunctionMetadata>();
+            int i = 0;
+            foreach (var route in routes)
+            {
+                var functionName = $"http-handler{i + 1}";
+                i++;
+
+                if (!TryValidateHttpRoute(route?.Route, out string error))
+                {
+                    AddFunctionError(functionName, error);
```

Comment:
I think `_errors` needs to be cleared between invocations of `CreateFunctionsFromRoutes`. I recommend creating an `ImmutableDictionary` builder as well in this method, and then building & assigning it to the `FunctionErrors` property at the end. This will address my other comment of the property getter recreating the immutable on each get, and also clear out the errors on each run.